designer/server/src/main/scala/pl/touk/nussknacker/ui/migrations/MigrationService.scala

```diff
@@ -0,0 +1,202 @@
+package pl.touk.nussknacker.ui.migrations
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.engine.api.process.{ProcessIdWithName, ProcessName}
+import pl.touk.nussknacker.engine.util.Implicits._
+import pl.touk.nussknacker.restmodel.scenariodetails.ScenarioParameters
+import pl.touk.nussknacker.restmodel.validation.ValidationResults.ValidationErrors
+import pl.touk.nussknacker.security.Permission
+import pl.touk.nussknacker.ui.api.MigrationApiEndpoints.Dtos.MigrateScenarioRequest
+import pl.touk.nussknacker.ui.api.{AuthorizeProcess, ListenerApiUser}
+import pl.touk.nussknacker.ui.listener.ProcessChangeEvent.OnSaved
+import pl.touk.nussknacker.ui.listener.{ProcessChangeEvent, ProcessChangeListener, User}
+import pl.touk.nussknacker.ui.process.ProcessService
+import pl.touk.nussknacker.ui.process.ProcessService.{
+  CreateScenarioCommand,
+  FetchScenarioGraph,
+  GetScenarioWithDetailsOptions,
+  UpdateScenarioCommand
+}
+import pl.touk.nussknacker.ui.process.migrate.{MigrationToArchivedError, MigrationValidationError}
+import pl.touk.nussknacker.ui.process.processingtype.ProcessingTypeDataProvider
+import pl.touk.nussknacker.ui.process.repository.ProcessRepository.RemoteUserName
+import pl.touk.nussknacker.ui.process.repository.UpdateProcessComment
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+import pl.touk.nussknacker.ui.uiresolving.UIProcessResolver
+import pl.touk.nussknacker.ui.validation.FatalValidationError
+import pl.touk.nussknacker.ui.{NuDesignerError, UnauthorizedError}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+trait MigrationService {
```

Comment:
let's remove the interface. When you don't know how to name the implementation and you add the "Impl" suffix, it almost always means that you don't need the interface at this moment.

Reply:
I tried to mimic what I've seen in the neighbour package(s), for example `pl.touk.nussknacker.ui.notifications.NotificationService`. Also I think that was suggested by @arkadius in his comment to this PR: 

"Let's extract a domain service for that instead of keeping everything in the infrastructure layer."

Reply:
For me @arkadius wants you to keep `MigrationApiService` as thin as possible. And it's good. But when your domain service has only one implementation (and you have no idea what the other implementation could look like), there is no sense in keeping the interface. That's why I propose to remove it (if you want to you can do the same for NotificationService trait)

Reply:
I agree. If there is only one implementation, the benefits of having the interface separated are lower than the drawbacks that it causes. Eventually, I would merge the interface with implementation in every place where we have such a situation to avoid confusion.

Reply:
I think we misunderstood each other. The `MigrationApiHttpService` was OK. We asked you to remove the `MigrationService` trait and rename the `MigrationServiceImpl` to `MigrationService`. That's all.